cdp_scrapers/legistar_utils.py

```diff
@@ -109,3 +184,805 @@ def get_legistar_events_for_timespan(
+        end: Optional[datetime] = datetime.utcnow(),
+    ) -> List[EventIngestionModel]:
+        """
+        Call get_legistar_events_for_timespan to retrieve Legistar API data
+        and return as List[EventIngestionModel]
+
+        Parameters
+        ----------
+        begin : datetime, default=datetime.utcnow() - timedelta(days=2)
+            By default query the past 2 days
+        end : datetime, default=datetime.utcnow()
+
+        Returns
+        -------
+        List[EventIngestionModel]
+            One instance of EventIngestionModel per Legistar API Event
+
+        See Also
+        --------
+        get_legistar_events_for_timespan
+        """
+        evs = []
+
+        for legistar_ev in get_legistar_events_for_timespan(
+            self.client_name,
+            begin=begin,
+            end=end,
+        ):
+            session_time = self.date_time_to_datetime(
+                legistar_ev[LEGISTAR_SESSION_DATE], legistar_ev[LEGISTAR_SESSION_TIME]
+            )
+
+            # prefer video file path in legistar Event.EventVideoPath
+            if legistar_ev[LEGISTAR_SESSION_VIDEO_URI] is not None:
+                list_uri = [
+                    {
+                        CDP_VIDEO_URI: stripped(
+                            legistar_ev[LEGISTAR_SESSION_VIDEO_URI]
+                        ),
+                        CDP_CAPTION_URI: None,
+                    }
+                ]
+            else:
+                list_uri = self.get_video_uris(legistar_ev) or [
+                    {CDP_VIDEO_URI: None, CDP_CAPTION_URI: None}
+                ]
+
+            sessions = []
+            sessions = reduced_list(
+                [
+                    self.get_none_if_empty(
+                        Session(
+                            session_datetime=session_time,
+                            session_index=len(sessions),
+                            video_uri=uri[CDP_VIDEO_URI],
+                            caption_uri=uri[CDP_CAPTION_URI],
+                        )
+                    )
+                    # Session per video
+                    for uri in list_uri
+                ]
+            )
+
+            ingested = self.get_none_if_empty(
+                EventIngestionModel(
+                    agenda_uri=stripped(legistar_ev[LEGISTAR_AGENDA_URI]),
+                    minutes_uri=stripped(legistar_ev[LEGISTAR_MINUTES_URI]),
+                    body=Body(name=stripped(legistar_ev[LEGISTAR_BODY_NAME])),
+                    sessions=sessions,
+                    event_minutes_items=self.get_event_minutes(
+                        legistar_ev[LEGISTAR_EV_ITEMS]
+                    ),
+                )
+            )
+            # let's not include None in the returned list
+            if ingested:
+                evs.append(ingested)
+
+        # TODO: better to return None if evs == [] ?
+        # return reduced_list(evs)
```

Comment:
I think you need to reduce the list here? There is a commented out reduced_list.

Also nah, list of len 0 is better than none as we just take this result and pass it directly to pipeline.